src/util.js

```diff
@@ -150,3 +150,32 @@ export function image_name(image) {
 export function is_valid_container_name(name) {
     return /^[a-zA-Z0-9][a-zA-Z0-9_\\.-]*$/.test(name);
 }
+
+export const validationClear = (validationFailed, key, onValidationChange) => {
+    if (!validationFailed)
+        return;
+
+    const delta = { ...validationFailed };
+    delete delta[key];
+    onValidationChange(delta);
+};
+
+export const dynamicValidate = (list, validationFailed, key, onValidationChange, validate) => {
+    const delta = validate(list, key, validationFailed);
+
+    onValidationChange(delta);
+};
+
+export const dynamicListOnValidationChange = (value, key, validationFailed, onValueChanged) => {
+    const validationFailedDelta = { ...validationFailed };
+    validationFailedDelta[key] = value;
+
+    if (validationFailedDelta[key].length == 0 || validationFailedDelta[key].every(a => a === undefined))
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20230725-122305-68cb803a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)